ceph/rbd/workflows/rbd.py

```diff
@@ -264,13 +273,500 @@ def create_pools_and_images(
+                ],
+                "get_time_taken": True,
+                "image_spec": [image_spec],
+                "operations": {
+                    "fs": "ext4",
+                    "io": True,
+                    "mount": True,
+                    "nounmap": False,
+                    "device_map": True,
+                },
+                "skip_mkfs": kw.get("skip_mkfs", False),
+            },
+        }
+        with parallel() as p:
+            p.spawn(krbd_io_handler, **io_config)
+            p.spawn(check_rbd_status, rbd=rbd, pool=pool, image=image, timeout=120)
+
+        out, err = rbd.status(pool=pool, image=image, format="json")
+        if err:
+            log.error(f"Error while fetching rbd status {pool}/{image}")
+            if test_ops_parallely:
+                raise Exception(f"Error while fetching rbd status {pool}/{image}")
+            return 1
+        rbd_status = json.loads(out)
+        log.info(f"RBD Status: {rbd_status}")
+        if rbd_status.get("watchers"):
+            log.error(
+                f"Watchers exist even while IOs are not running for {pool}/{image}"
+            )
+            if test_ops_parallely:
+                raise Exception(
+                    f"Watchers exist even while IOs are not running for {pool}/{image}"
+                )
+            return 1
+    except Exception as e:
+        if test_ops_parallely:
+            raise Exception(
+                f"Run IOs and verify rbd status failed for {pool}/{image} with error {e}"
+            )
+        return 1
+
+    return 0
+
+
+def wrapper_for_image_ops(**kw):
+    """
+    Perform specified image operation for images in pool,
+    and verify either sequentially or parallely based on input
+
+    Args:
+        kw: {
+            "rbd": <>,
+            "pool": <>,
+            "image_config": {
+                <image_1>: <image1_conf>,
+                <image_2>: <image2_conf>,....
+            },
+            "ops_module": <>,
+            "ops_method": <>,
+            "test_ops_parallely": <>
+        }
+    """
+    pool = kw.get("pool")
+    image_config = kw.pop("image_config", {})
+    test_ops_parallely = kw.get("test_ops_parallely", False)
+    ops_module = kw.pop("ops_module", {})
+    mod_obj = import_module(ops_module)
+    ops_method = kw.pop("ops_method", {})
+    method_obj = getattr(mod_obj, ops_method)
+    log.info(f"Test image operations {ops_method} for pool {pool}")
+
+    if test_ops_parallely:
+        with parallel() as p:
+            for image, image_conf in image_config.items():
+                p.spawn(method_obj, image=image, image_conf=image_conf, **kw)
+    else:
+        for image, image_conf in image_config.items():
+            rc = method_obj(image=image, image_conf=image_conf, **kw)
+            if rc:
+                log.error(
+                    f"Test image feature enable and disable failed for {pool}/{image}"
```

Comment:
error message needs to be image operation better we can log what operation is it as well.

Reply:
failed.